src/Common/HashTable/Hash.h

```diff
@@ -48,33 +48,36 @@ inline DB::UInt64 intHash64(DB::UInt64 x)
 #include <arm_acle.h>
 #endif
 
-inline DB::UInt64 intHashCRC32(DB::UInt64 x)
+/// NOTE: Intel intrinsic can be confusing.
+/// - https://code.google.com/archive/p/sse-intrinsics/wikis/PmovIntrinsicBug.wiki
+/// - https://stackoverflow.com/questions/15752770/mm-crc32-u64-poorly-defined
+inline DB::UInt32 intHashCRC32(DB::UInt64 x)
```

Comment:
Looks unsafe, and I guess it is better to keep signature as before and fix the callers. Thoughts?

Reply:
Yes, that looks unsafe, please keep the signature.

Reply:
Actually there was not too much with previous signature, will be fixed shortly (need to fix conflicts with upstream, sigh).